src/ndtensor.zig

```diff
@@ -95,6 +95,33 @@ pub fn NDTensor(comptime T: type) type {
             return self;
         }
 
+        pub fn zeros(shape: []const usize, requires_grad: bool, device: DeviceReference) !*Self {
+            const self = try device.allocator.create(Self);
+            self.* = Self{
+                .data = try dtype.zeros(shape, device),
+                .grad = if (requires_grad) try dtype.zeros(shape, device) else null,
+                .requires_grad = requires_grad,
+                .device = device,
+            };
+            return self;
+        }
+
+        pub fn getShape(self: Self) []const usize {
```

Comment:
I had to prefix these with "get" because we are getting name collisions otherwise.

Reply:
lgtm